src/routines/mpilib20_init_finalise.f90

```diff
@@ -4,33 +4,55 @@
 !>  to be used as free subroutines.
 module mpilib20_init_finalise
   use, intrinsic :: iso_fortran_env, only: error_unit
-  use mpi_bindings, only: MPI_comm, MPI_Group
+  use mpi_bindings, only : MPI_Comm, MPI_Group, MPI_INIT, MPI_COMM_DUP, &
+                           MPI_INIT_THREAD, MPI_COMM_RANK, MPI_ABORT, MPI_COMM_DUP, &
+                           MPI_COMM_SIZE, MPI_GROUP_SIZE, MPI_FINALIZE, MPI_THREAD_SINGLE, &
+                           MPI_THREAD_FUNNELED, MPI_THREAD_SERIALIZED,  MPI_THREAD_MULTIPLE    
   implicit none
   private 
 
-  !> Process designated as master/root 
-  integer, public, parameter :: root_id = 0
-  
-  ! TODO. Add get and set 
   !> MPI environment type
   type, public :: mpi_env_type
-     type(MPI_comm)  :: comm         !! MPI communicator (integer in older bindings)
-     integer         :: process      !! Process id (rank)
-     integer         :: n_processes  !! Total number of processes
-     type(MPI_Group) :: group        !! Group id (integer in older bindings)
-     integer         :: group_size   !! Number of processes in group
-     integer         :: ierror       !! Error code
-     
-   contains
-    procedure :: init => init_mpi_env         !! Initialise instance of MPI environment object
-    procedure :: finalize => finalise_mpi_env !! Terminate instance of MPI environment object
+     !private  TODO(Alex) Make data private. Add getters and setters 
+       
+     integer,public        :: root_id      !! Process designated as master/root 
+     type(MPI_Comm), public  :: comm         !! MPI communicator (integer in older bindings)
+     integer, public        :: process      !! Process id (rank)
+     integer,public         :: n_processes  !! Total number of processes
+     type(MPI_Group), public :: group        !! Group id (integer in older bindings)
+     integer, public         :: group_size   !! Number of processes in group
+     integer, public         :: ierror       !! Error code
 
+   contains
+    private
+    ! Public routines 
+    procedure, public :: init     => init_mpi_env         !! Initialise instance of MPI environment object
+    procedure, public :: finalize => finalise_mpi_env     !! Terminate instance of MPI environment object
+    procedure, public :: get_comm => get_communicator     !! Get the communicator 
+    ! Private routines
+    !procedure  :: set_comm  !! Set the communicator... may not need
```

Comment:
Not sure if private data can be accessed within the module in which the type is defined. Would assume not but then this is fortran.